### PR TITLE
Go: sync FlowSummaryImpl.qll

### PR DESCRIPTION
**Pull request by asgerf**
I was trying to do a Go version of https://github.com/github/codeql/pull/7878, but it seem the `FlowSummaryImpl.qll` file is out of sync with `github/codeql`.

This updates the `FlowSummaryImpl.qll` to the version currently on `github/codeql`, and manually adds the need classes and predicates to `FlowSummaryImplSpecific.qll`.

### CODE REVIEW

**Comment by owen-mc** [comment]
We normally sync all the shared data flow library files at once by running `make sync-dataflow-libraries`.

**Comment by owen-mc** [comment]
@asgerf It would just be more work for you, and the risk of delaying getting these changes in. There might be more language-specific additions needed, and we'd want to do a performance evaluation before merging.

File: ql/lib/semmle/go/dataflow/internal/FlowSummaryImplSpecific.qll [hunk 1]
```diff
@@ -254,3 +286,20 @@ predicate parseReturn(string c, int n) {
...
   )
 }
+
+bindingset[arg]
+private int parseConstantOrRange(string arg) {
+  result = arg.toInt()
+  or
+  exists(int n1, int n2 |
+    arg.regexpCapture("ReturnValue\\[([-0-9]+)\\.\\.([0-9]+)\\]", 1).toInt() = n1 and
+    arg.regexpCapture("ReturnValue\\[([-0-9]+)\\.\\.([0-9]+)\\]", 2).toInt() = n2 and
```
Comment (owen-mc) [inline ql/lib/semmle/go/dataflow/internal/FlowSummaryImplSpecific.qll:296]: I think this shouldn't have the `ReturnValuex\\[` and `\\]` parts. And then you should use it in `ParseReturn`.

Comment (asgerf) [reply ql/lib/semmle/go/dataflow/internal/FlowSummaryImplSpecific.qll:296]: Oh this was silly copy-pasta. I'm slightly worried about the tests still passing 🤔. I'll try and add a test that would have caught this.

Comment (owen-mc) [reply ql/lib/semmle/go/dataflow/internal/FlowSummaryImplSpecific.qll]: Thanks for improving the test. Multiple return values are mainly used to return a value plus an error, so we didn't think to test the `ReturnValue[n1..n2]` syntax.

**Comment by owen-mc** [comment]
This looks good. I'll just do a performance evaluation before approving and merging.

**Comment by asgerf** [comment]
[Evaluation](https://github.com/github/codeql-dca-main/tree/data/asgerf/sync-flow-summa__Differences__code-scanning/reports) of the original PR looked fine.

**Review by owen-mc — APPROVED** [review]

**Pull request merged**